packages/camera/camera_platform_interface/lib/src/types/video_capture_options.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'camera_image_data.dart';
+
+/// Options wrapper for [CameraPlatform.startVideoCapturing] parameters.
+@immutable
+class VideoCaptureOptions {
+  /// Constructs a new instance.
+  const VideoCaptureOptions(
+    this.cameraId, {
+    this.maxDuration,
+    this.streamCallback,
+    this.streamOptions,
+  });
+
+  /// The ID of the camera to use for capturing.
+  final int cameraId;
+
+  /// The maximum time to perform capturing for.
+  ///
+  /// By default there is no maximum on the capture time.
+  final Duration? maxDuration;
+
+  /// An optional callback to enable streaming.
+  /// If set, then each image captured by the camera will be
+  /// passed to this callback.
+  final Function(CameraImageData image)? streamCallback;
+
+  /// Configuration options for streaming.
+  /// Should only be set if a streamCallback is also present.
```

Comment:
If this is true, can you add an assertion to the constructor that verifies this?

Reply:
Sure, done.